app/controllers/web_requests_controller.rb

```diff
@@ -24,7 +26,13 @@ def handle_request
     if user
       agent = user.agents.find_by_id(params[:agent_id])
       if agent
-        content, status, content_type = agent.trigger_web_request(request)
+        content, status, content_type, headers = agent.trigger_web_request(request)
+
+        if headers.presence
+          headers.each do |k,v|
+            response.headers[k] = v
+          end
+        end
```

Comment:
I agree, I think we have to trust the Agent implementer / user. (But we do need to be careful to avoid event parameters masking options in some hypothetical Agent. E.g.,  `options.merge(event)` could be dangerous because the event might be coming from an external source, like a website or post API.)